src/Relation/Binary/Construct/Interior/Symmetric.agda

```diff
@@ -0,0 +1,91 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Symmetric interior of a binary relation
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Relation.Binary.Construct.Interior.Symmetric where
+
+open import Function.Base using (flip)
+open import Level
+open import Relation.Binary
+
+private
+  variable
+    a b c ℓ r s t : Level
+    A : Set a
+    R : Rel A r
+    S : Rel A s
+    T : Rel A t
+
+------------------------------------------------------------------------
+-- Definition
+
+record SymInterior (R : Rel A ℓ) (x y : A) : Set ℓ where
+  constructor _,_
+  field
+    lhs≤rhs : R x y
+    rhs≤lhs : R y x
+open SymInterior public
+
+------------------------------------------------------------------------
+-- Properties
+
+-- The symmetric interior is symmetric.
+symmetric : Symmetric (SymInterior R)
+symmetric (r , r′) = r′ , r
+
+-- The symmetric interior of R is greater than (or equal to) any other symmetric
+-- relation contained by R.
+unfold : Symmetric S → S ⇒ R → S ⇒ SymInterior R
+unfold sym f s = f s , f (sym s)
+
+-- SymInterior preserves various properties.
+reflexive : Reflexive R → Reflexive (SymInterior R)
+reflexive refl = refl , refl
+
+transitive′ : Trans R S T → Trans S R T →
+  Trans (SymInterior R) (SymInterior S) (SymInterior T)
+transitive′ trans trans′ (r , r′) (s , s′) = trans r s , trans′ s′ r′
+
+transitive : Transitive R → Transitive (SymInterior R)
+transitive {R = R} tr = transitive′ {R = R} tr tr
+
+-- The symmetric interior of a strict relation is empty.
+Empty-SymInterior : Asymmetric R → Empty (SymInterior R)
+Empty-SymInterior asym (r , r′) = asym r r′
```

Comment:
`asymmetric⇒empty`?

Also, I think it's not always straightforward as to whther you announce a proposition/predicate instance/relation instance as being empty, or whether you prove it implies/is contained in, the corresponding bottom element `Empty` of the ordering... and here I do confess I'm not clear either what policy the library has, and if it does, what names are recommended here...

Reply:
This looks like another case of the module name necessarily being part of the name (because asymmetric relations are not necessarily empty). Somehow it doesn't feel like the right thing to do in this case, when it's just a single property.

As for `Empty`, this is a new definition in this PR, so there's no convention *to* use it. I hadn't thought of `_⇒ Never`, but I think it's definitionally equal anyway. `Never` also appears to be unused in the library. It feels similar to the difference between `P → ⊥` and `¬ P`.

Reply:
I strongly agree with @jamesmckinna on this one. None of the other results in this file (or similar files) have the name of the relation in the proof.